File: libs/document-processor/src/__tests__/models/set-property-on-event.ts
import { z } from 'zod';
import { withTypeBlueId } from '@blue-labs/language';

import { handlerContractBaseSchema } from '../../model/shared/index.js';

export const setPropertyOnEventSchema = withTypeBlueId('SetPropertyOnEvent')(
  handlerContractBaseSchema.extend({
    expectedKind: z.string().optional(),
    propertyKey: z.string(),
    propertyValue: z.number().int(),
  }),
);

export type SetPropertyOnEvent = z.infer<typeof setPropertyOnEventSchema>;

File: references/java-sdk/src/test/java/blue/language/processor/model/SetPropertyOnEvent.java
package blue.language.processor.model;

import blue.language.model.TypeBlueId;
import blue.language.processor.model.HandlerContract;

@TypeBlueId("SetPropertyOnEvent")
public class SetPropertyOnEvent extends HandlerContract {

    private String expectedKind;
    private String propertyKey;
    private int propertyValue;

    public String getExpectedKind() {
        return expectedKind;
    }

    public SetPropertyOnEvent expectedKind(String expectedKind) {
        this.expectedKind = expectedKind;
        return this;
    }

    public String getPropertyKey() {
        return propertyKey;
    }

    public SetPropertyOnEvent propertyKey(String propertyKey) {
        this.propertyKey = propertyKey;
        return this;
    }

    public int getPropertyValue() {
        return propertyValue;
    }

    public SetPropertyOnEvent propertyValue(int propertyValue) {
        this.propertyValue = propertyValue;
        return this;
    }
}
